From: Andrew Morton <akpm@linux-foundation.org>
Date: Thu, 6 Aug 2026 14:14:15 -0700
Subject: [PATCH] 
 mm-page_alloc-only-update-lowmem_reserve_ratio-on-sysctl-write-fix

add comment, per hannes

Link: https://lore.kernel.org/anSRGASe5FIrqwlg@cmpxchg.org
Cc: Brendan Jackman <jackmanb@google.com>
Cc: Jianlin Shi <shijianlin11@foxmail.com>
Cc: Johannes Weiner <hannes@cmpxchg.org>
Cc: Michal Hocko <mhocko@suse.com>
Cc: Suren Baghdasaryan <surenb@google.com>
Cc: Vlastimil Babka <vbabka@kernel.org>
Cc: Zi Yan <ziy@nvidia.com>
Cc: Joel Granados <joel.granados@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 mm/page_alloc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/mm/page_alloc.c b/mm/page_alloc.c
index 408364ff2c93a..5d4f1a1ed4b4e 100644
--- a/mm/page_alloc.c
+++ b/mm/page_alloc.c
@@ -6932,6 +6932,10 @@ static int lowmem_reserve_ratio_sysctl_handler(const struct ctl_table *table,
 	if (!write)
 		return proc_dointvec_minmax(table, write, buffer, length, ppos);
 
+	/*
+	 * proc_dointvec_max() works incrementally. Use a buffer and only set
+	 * the values if all of them parse cleanly.
+	 */
 	memcpy(ratio, sysctl_lowmem_reserve_ratio, sizeof(ratio));
 	tmp.data = ratio;
 
-- 
2.47.3

